src/init/features/project.ts

```diff
@@ -0,0 +1,149 @@
+  }
+  const projectId = project.projectId;
+  const name = project.displayName;
+  return {
+    id: projectId,
+    label: `${projectId} (${name})`,
+    instance: _.get(project, "resources.realtimeDatabaseInstance"),
+  };
+}
+
+/**
+ * Presents user with list of projects to choose from and gets project
+ * information for chosen project.
+ * @param options Command line options.
+ * @returns A {@link FirebaseProject} object.
+ */
+async function selectProjectFromList(options: any): Promise<ProjectInfo> {
+  let project: FirebaseProject | undefined;
+  const projects: FirebaseProject[] = await listProjects();
+  let choices = projects.filter((p: FirebaseProject) => !!p).map((p) => {
+    return {
+      name: `${p.projectId} (${p.displayName})`,
+      value: p.projectId,
+    };
+  });
+  choices = _.orderBy(choices, ["name"], ["asc"]);
+  choices.unshift({ name: NO_PROJECT, value: NO_PROJECT });
+  choices.push({ name: NEW_PROJECT, value: NEW_PROJECT });
+
+  if (choices.length >= 25) {
+    utils.logBullet(
+      `Don't want to scroll through all your projects? If you know your project ID, ` +
+        `you can initialize it directly using ${clc.bold(
+          "firebase init --project <project_id>"
+        )}.\n`
+    );
+  }
+  const projectId: string = await promptOnce({
+    type: "list",
+    name: "id",
+    message: "Select a default Firebase project for this directory:",
+    validate: (answer: any) => {
+      if (!_.includes(choices, answer)) {
+        return `Must specify a Firebase project to which you have access.`;
+      }
+      return true;
+    },
+    choices,
+  } as Question);
+  if (projectId === NEW_PROJECT || projectId === NO_PROJECT) {
+    return { id: projectId };
+  }
+
+  project = projects.find((p) => p.projectId === projectId);
+  const pId = choices.find((p) => p.value === projectId);
+  const label = pId ? pId.name : "";
+
+  return {
+    id: projectId,
+    label,
+    instance: _.get(project, "resources.realtimeDatabaseInstance"),
+  };
+}
+
+/**
+ * Sets up the default project if provided and writes .firebaserc file.
+ * @param setup A helper object to use for the rest of the init features.
+ * @param config Configuration for the project.
+ * @param options Command line options.
+ */
+export async function doSetup(setup: any, config: Config, options: any): Promise<void> {
+  setup.project = {};
+
+  logger.info();
+  logger.info(`First, let's associate this project directory with a Firebase project.`);
+  logger.info(
+    `You can create multiple project aliases by running ${clc.bold("firebase use --add")}, `
+  );
+  logger.info(`but for now we'll just set up a default project.`);
+  logger.info();
+
+  if (_.has(setup.rcfile, "projects.default")) {
+    utils.logBullet(`.firebaserc already has a default project, skipping`);
+    setup.projectId = _.get(setup.rcfile, "projects.default");
+    return;
+  }
+
+  const projectInfo: ProjectInfo = await getProjectInfo(options);
```

Comment:
Is it necessary to say `const projectInfo: ProjectInfo`? can the `ProjectInfo` be dropped?